keystrokes/src/components/Character/CharacterLists/CharacterList.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+import styleGuide from '../../../stylesheets/style';
+import { ref } from '../../../config/constants';
+
+CharacterList.propTypes = {
+	userInfo: PropTypes.object.isRequired,
+	userPath: PropTypes.string.isRequired,
+};
+
+function CharacterList({ userInfo, userPath, ...props }) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState('');
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		setCharacters(userInfo.characters || {});
+		setCurrentCharacter(userInfo.currentCharacter);
+	}, [userInfo]);
+
+	async function handleSubmit(fields) {
+		const name = fields.name;
+		const key = ref.child('/Characters').push().key;
+		setCurrentCharacter(key);
+		setCharacters({ ...characters, [key]: { name } });
+		props.history.push(
+			`${props.location.pathname}/${key}/${JSON.stringify(fields)}`
+		);
+	}
+
+	return (
+		<React.Fragment>
+			<CharacterCreationModal
+				isOpen={open}
+				setIsOpen={setOpen}
+				handleSubmit={handleSubmit}
+			/>
+			<Typography variant="h1" align="center">
+				Character List
+			</Typography>
+			<Typography
+				variant="h3"
+				align="center"
+				color="textSecondary"
+				gutterBottom
+			>
+				Edit a character from the list or create a new character.
+			</Typography>
+			<div className={classes.characterList}>
+				<List>
+					{[
+						...Object.keys(characters).sort((a, b) =>
```

Comment:
just fyi sort mutates the original object, looks like you're trying to use the spread operator to copy it. Just worth knowing the original is also changed.

Reply:
any reason you're sorting by keys? Check out https://lodash.com/docs/#sortBy

Reply:
Good call about the spread, i think that was leftover from trying to get selected character first. Which is why i sort, im trying to make selected character be at the top of the list (currently there is no way to select a character... but when there is this will be useful!)